packages/gestalt/src/List/getChildrenToArray.js

```diff
@@ -0,0 +1,50 @@
+// @flow strict
+import { cloneElement, Fragment, Children, type Node } from 'react';
+
+const ALLOWED_CHILDREN_MAP = {
+  List: 'List.Item',
+  NestedList: 'List.Item',
```

Comment:
This doesn't seem correct?

Reply:
It's correct

Reply:
Ah, I think the structure of this map is unintuitive. I didn't realize the keys represented parent components. Perhaps something like `allowedChildrenByParent` would be more clear?

Reply:
DONE